### PR TITLE
ported pose2twist from ROS1 to ROS2

### PR DESCRIPTION
**Pull request by nik-tier4**
This PR ports pose2twist from ROS1 to ROS2

Changes and notes I made during the port.

The naming of header files are all small letter in Ros2:
`<std_msgs/Float32.h> to <std_msgs/msg/float32.hpp>`
`geometry_msgs/PoseStamped.h to geometry_msgs/msg/pose_stamped.hpp`

### CODE REVIEW

**Review by fred-apex-ai — CHANGES_REQUESTED** [review]

File: localization/twist_estimator/pose2twist/CMakeLists.txt [hunk 1]
```diff
@@ -1,43 +1,53 @@
...
-  roscpp
-  tf2_geometry_msgs
-  std_msgs
-  geometry_msgs
-)
-
-catkin_package(
-  INCLUDE_DIRS include
-)
+find_package(ament_cmake REQUIRED)
```
[the remark this answers is not in the document]
Comment (nik-tier4) [reply localization/twist_estimator/pose2twist/CMakeLists.txt]: yes, updated it.

File: localization/twist_estimator/pose2twist/include/pose2twist/pose2twist_core.h [hunk 2]
```diff
@@ -16,28 +16,26 @@
 
 #pragma once
```
[the remark this answers is not in the document]
Comment (nik-tier4) [reply localization/twist_estimator/pose2twist/include/pose2twist/pose2twist_core.h:17]: never used linter before, so what is the change needed?

**Review by fred-apex-ai — COMMENTED** [review]
changes fix my comments. Thans

**Comment by mitsudome-r** [comment]
@nik-tier4 Can you rebase your branch to newest ros2 branch?
```
git fetch origin
git rebase origin/ros2
```

**Review by nik-tier4 — COMMENTED** [review]
include guards fixed

**Pull request merged**